### PR TITLE
[Gardening] Use string constants for Builtin module and type names

### PR DESCRIPTION
**Pull request by kitasuke**
I defined new string constants in `String.h` for Builtin so that we could replace string value with it.

- [x] Added `BUILTIN_NAME`
- [x] Added `BUILTIN_TYPE_NAME`s

~It might be a bit different case, but should I also apply to Builtin type name like here?~
~https://github.com/apple/swift/blob/master/lib/IDE/TypeReconstruction.cpp#L795:L797~

Here is a link for the discussion.
https://github.com/apple/swift/pull/14959#discussion_r172034272

### CODE REVIEW

**Comment by omochi** [comment]
I think we should fix `TypeReconstruction.cpp` too.

**Comment by harlanhaskins** [comment]
@swift-ci please smoke test

**Comment by kitasuke** [comment]
@harlanhaskins I also added `BUILTIN_TYPE_NAME`s. Let me know if I should break constant string file into specific one for type name.

File: lib/AST/ASTPrinter.cpp [hunk 1]
```diff
@@ -3228,21 +3228,23 @@ class TypePrinter : public TypeVisitor<TypePrinter> {
         llvm::raw_svector_ostream UnderlyingOS(UnderlyingStrVec);
         T->getElementType().print(UnderlyingOS);
       }
-      if (UnderlyingStrVec.startswith("Builtin."))
+      std::string builtin_name = BUILTIN_NAME;
+      std::string builtin_type_prefix = builtin_name + ".";
```
Comment (harlanhaskins) [inline lib/AST/ASTPrinter.cpp]: Since we dynamically construct this a few times, we should maybe just make a new constant for “Builtin.”

**Review by harlanhaskins — CHANGES_REQUESTED** [review]
LGTM once we factor out “Builtin.”

Thanks for this!

File: lib/AST/ASTPrinter.cpp (hunk 1 above)
Comment (kitasuke) [reply lib/AST/ASTPrinter.cpp]: Good point. Fixed it.

**Comment by kitasuke** [comment]
@harlanhaskins Addressed your feedback! Would be appreciated if you could review again!

**Review by harlanhaskins — COMMENTED** [review]
Sorry, I meant to submit this comment several days ago 😓

File: include/swift/Strings.h [hunk 2]
```diff
@@ -35,13 +35,52 @@ namespace swift {
...
+  /// The name of the Builtin type for Int64
+  static const char BUILTIN_TYPE_NAME_INT64[] = "Builtin.Int64";
+  /// The name of the Builtin type for Int128
+  static const char BUILTIN_TYPE_NAME_INT128[] = "Builtin.Int128";
+  /// The name of the Builtin type for Int256
+  static const char BUILTIN_TYPE_NAME_INT256[] = "Builtin.Int256";
+  /// The name of the Builtin type for Int512
+  static const char BUILTIN_TYPE_NAME_INT512[] = "Builtin.Int512";
+  /// The name of the Builtin type for Float
+  static const char BUILTIN_TYPE_NAME_Float[] = "Builtin.Float";
```
[the remark this answers is not in the document]
Comment (kitasuke) [reply include/swift/Strings.h]: Sorry for trivial mistake... Fixed it.

**Comment by kitasuke** [comment]
I squashed and pushed the commit, thanks for reviewing!

**Review by harlanhaskins — APPROVED** [review]

**Comment by harlanhaskins** [comment]
LGTM!

@swift-ci please smoke test and merge

**Comment by harlanhaskins** [comment]
Thanks @kitasuke!

**Pull request merged**